services/command-router/src/main/java/org/eclipse/hono/commandrouter/impl/pubsub/PubSubBasedCommandConsumerFactoryImpl.java

```diff
@@ -0,0 +1,221 @@
+    private final PubSubSubscriberFactory subscriberFactory;
+    private final Set<String> tenantIds = new HashSet<>();
+    private final Map<String, MessageReceiver> activeReceivers = new ConcurrentHashMap<>();
+    private PubSubBasedMappingAndDelegatingCommandHandler commandHandler;
+
+    /**
+     * Creates a new factory to process commands via Google Pub/Sub.
+     *
+     * @param vertx The Vert.x instance to use.
+     * @param tenantClient The Tenant service client.
+     * @param tracer The tracer instance.
+     * @param pubSubPublisherFactory The publisher factory for creating Pub/Sub publishers for sending messages.
+     * @param projectId The identifier of the Google Cloud Project to connect to.
+     * @param commandTargetMapper The component for mapping an incoming command to the gateway (if applicable) and
+     *            protocol adapter instance that can handle it. Note that no initialization of this factory will be done
+     *            here, that is supposed to be done by the calling method.
+     * @param metrics The component to use for reporting metrics.
+     * @param subscriberFactory The subscriber factory for creating Pub/Sub subscribers for receiving messages.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     */
+    public PubSubBasedCommandConsumerFactoryImpl(
+            final Vertx vertx,
+            final TenantClient tenantClient,
+            final Tracer tracer,
+            final PubSubPublisherFactory pubSubPublisherFactory,
+            final String projectId,
+            final CommandTargetMapper commandTargetMapper,
+            final CommandRouterMetrics metrics,
+            final PubSubSubscriberFactory subscriberFactory) {
+        this.vertx = Objects.requireNonNull(vertx);
+        this.tenantClient = Objects.requireNonNull(tenantClient);
+        this.tracer = Objects.requireNonNull(tracer);
+        this.commandTargetMapper = Objects.requireNonNull(commandTargetMapper);
+        this.metrics = Objects.requireNonNull(metrics);
+        this.subscriberFactory = Objects.requireNonNull(subscriberFactory);
+        Objects.requireNonNull(pubSubPublisherFactory);
+        Objects.requireNonNull(projectId);
+
+        this.internalCommandSender = new PubSubBasedInternalCommandSender(pubSubPublisherFactory, projectId, tracer);
+        this.responseSender = new PubSubBasedCommandResponseSender(vertx, pubSubPublisherFactory, projectId, tracer);
+    }
+
+    @Override
+    public Future<Void> start() {
+        if (lifecycleStatus.isStarting()) {
+            return Future.succeededFuture();
+        } else if (!lifecycleStatus.setStarting()) {
+            return Future.failedFuture(new IllegalStateException("factory is already started/stopping"));
+        }
+        if (Vertx.currentContext() == null) {
+            return Future.failedFuture(new IllegalStateException("factory must be started in a Vert.x context"));
+        }
```

Comment:
It looks like there is no way to create an instance of this class without providing a Vertx instance so IMHO this check is not necessary but rather misleading ...